sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationServiceSerializer.cs

```diff
@@ -13,74 +13,127 @@ namespace Azure.Data.AppConfiguration
-            json.WriteString("value", setting.Value);
-            json.WriteString("content_type", setting.ContentType);
+            WriteRequestBody(json, setting);
+            json.WriteEndObject();
+            json.Flush();
+            return writer.WrittenMemory;
+        }
+
+        public static void WriteRequestBody(Utf8JsonWriter writer, ConfigurationSetting setting)
+        {
+            writer.WriteString("value", setting.Value);
+            writer.WriteString("content_type", setting.ContentType);
             if (setting.Tags != null)
             {
-                json.WriteStartObject("tags");
+                writer.WriteStartObject("tags");
                 foreach (System.Collections.Generic.KeyValuePair<string, string> tag in setting.Tags)
                 {
-                    json.WriteString(tag.Key, tag.Value);
+                    writer.WriteString(tag.Key, tag.Value);
                 }
-                json.WriteEndObject();
+
+                writer.WriteEndObject();
             }
+
             if (setting.ETag != default)
             {
-                json.WriteString("etag", setting.ETag.ToString());
+                writer.WriteString("etag", setting.ETag.ToString());
             }
-            json.WriteEndObject();
-            json.Flush();
+        }
+
+        public static ConfigurationSetting ReadSetting(ref Utf8JsonReader reader)
+        {
+            using JsonDocument json = JsonDocument.ParseValue(ref reader);
+            JsonElement root = json.RootElement;
+            return ReadSetting(root);
         }
 
         private static ConfigurationSetting ReadSetting(JsonElement root)
         {
             // TODO (pri 2): make the deserializer version resilient
             var setting = new ConfigurationSetting();
-            if (root.TryGetProperty("key", out JsonElement keyValue))
-                setting.Key = keyValue.GetString();
-            if (root.TryGetProperty("value", out JsonElement value))
-                setting.Value = value.GetString();
-            if (root.TryGetProperty("label", out JsonElement labelValue))
-                setting.Label = labelValue.GetString();
-            if (root.TryGetProperty("content_type", out JsonElement contentValue))
-                setting.ContentType = contentValue.GetString();
-            if (root.TryGetProperty("etag", out JsonElement eTagValue))
-                setting.ETag = new ETag(eTagValue.GetString());
-            if (root.TryGetProperty("last_modified", out JsonElement lastModified))
-            {
-                if (lastModified.ValueKind == JsonValueKind.Null)
+            foreach (JsonProperty property in root.EnumerateObject())
+            {
+                ReadPropertyValue(setting, property);
+            }
+            return setting;
+        }
+
+        private static void ReadPropertyValue(ConfigurationSetting setting, JsonProperty property)
+        {
+            if (property.NameEquals("content_type"))
```

Comment:
Would a switch statement make sense here?

Reply:
`switch` is implemented either as a bunch of `if/else` statements or as a dictionary, depending on the amount of the strings in the switch statement. The benefit of `NameEquals` is that it tries to compare string to a property name without memory allocation: https://github.com/dotnet/corefx/blob/7cae92b39d2fed2e4e88e900f2e7d787ed9f6cfa/src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs#L285